### ISSUE TITLE
【ios】utils::PostconditionPanic: (null) libc++abi.dylib __ZSt11__terminatePFvvE . Need Help!!!

### ISSUE THREAD

**Issue by dandingol03** [google/filament#5002]
**Describe the bug**
crash on iPhone 13 Pro 、 iPhone 13 Pro Max、iPhone 11、iPhone 10

**To Reproduce**
crashed randomly

**Logs**
```
#0 FEngine::loop
utils::PostconditionPanic
(null)
```

**Smartphone (please complete the following information):**
 - Device: [iPhone]
 - OS: [e.g. 13、13 pro 、11、10]

**Additional context**
Does Anyone have idea?

**Comment by bejado** [comment]
Are you running a particular iOS sample?

**Comment by dandingol03** [comment]
HI, the answer is **yes**.I  run my own demo which is based on the official `hello-pbr` demo . However, there is something different from the `hello-pbr` demo :

1. I use `UIIMage` and `CFDataRef` to decode the jpg file which is needed by creating texture
2. I release `CFDataRef` in the `PixelBufferDescriptor` callback.
3. I initialize a `CVPixelBuffer` for recording video, but i never use it 
4. There is a enter animation added when i enter the rendering scene.

**additional contex**
 I use an AB test to keep the code affects  the user online. Until now, there is 3000 users use the engine, and the crash happened 9 times.

**Comment by bejado** [comment]
If you can upload a complete sample that reproduces the issue consistently, I can take a look.

**Comment by dandingol03** [comment]
Sorry, i can't reproduce this problem consistently. There is almost no description  about the crash printed, close it first.

**Comment by dandingol03** [comment]
I still can't figure out the reason why it crashed and no message useful to print. If i set the `Panic::Mode` to throw, the crash will not happen? I just want to know how to fix or avoid this weried crash. I will upload my code later, open this question again~

**Comment by bejado** [comment]
What backend are you using? Is this the same issue as https://github.com/google/filament/issues/5027?

**Comment by dandingol03** [comment]
No. These two question is different, i use `metal` backend.

**Comment by bejado** [comment]
Are you able to upload your code? You could also try with a debug build of Filament to get a more verbose error message.

**Comment by dandingol03** [comment]
I think the (null) message is not reported by `filament`. It is reported by the crash logger developed by the company i work at, and it can only report the crash stack. So the crash logger i uses cannot report the detail message about `PostConditionPanic`. After i fix a crash happened in ios device, the crash with (null) message is obviously less. By the way the crash i fix also happened in android device, but the crash in android device can report detail message. 
  In conclusion, this crash with (null) message is probably made up of some other crash, and the message report nothing because the crash logger i use cannot catch `filament` `panic`. Close this problem,  because the crash with `(null)` message is not related to `filament`. Thanks your time~ @bejado
